c/client.c

```diff
@@ -105,3 +105,11 @@ void free_wrapped_context(wrapped_context_t *wctx) {
     free(wctx);
   }
 }
+
+void *malloc_or_fail(size_t len) {
+  void *block = malloc(len);
+  if (block == NULL) {
+    abort();
```

Comment:
I wonder if `abort()` is a good idea for a library. For a "program", abort is probably fine, but for a library, we might want to return an error, so that the user of the library can show an error message and quit.

Reply:
if you are going to take that route of indicating the error: I don't know what the best way to indicate an error would be. May be return an error (`int`) but pass a pointer to a pointer or something for the `malloc()` to return back? Or may be do everything in the Go side itself? 

```
		c_buffer := C.malloc(MAX_READ_BUFFER_LEN)
                if c_buffer == nil {
                        return errors.New(....)
                }
```
something of this sort? `recvFile()` would need to return an `error` type in that case.. 

But I leave it to you to decide if you want to go this route or just use `abort()`.

Reply:
I have added a handler to the context(to be provided by a client) which would then be called whenever malloc returns NULL and tries to signal the transfer failure gracefully after that.